app/models/piece.rb

```diff
@@ -48,4 +48,32 @@ def valid_move?(x_destination, y_destination)
   def diff(destination, current)
     (destination - current).abs
   end
+
+
+  # Capture logic
+  # returns true if the move is valid
+  def move_to!(x_destination, y_destination)
+    return true if valid_move?(x_destination, y_destination)
```

Comment:
Think this would need to be `return false unless valid_move?(x_destination, y_destination)`, so that the rest of the method is executed when it's a valid move?